src/dispatcher.rs

```diff
@@ -92,17 +100,25 @@ impl TxnDispatcher for Dispatcher {
         &self,
         batch: &TxnBatch,
         mut ctx: Context,
-    ) -> (TxnBatch, Vec<Tags>, Vec<RoothashMessage>) {
+    ) -> Result<(TxnBatch, Vec<Tags>, Vec<RoothashMessage>)> {
         // Invoke start batch handler.
         self.batch_handler.start_batch(&mut ctx);
 
         #[cfg(feature = "prefetch")]
         let mut prefixes: Vec<Prefix> = Vec::new();
 
         // Decode and check transactions in this batch.
-        let calls: Vec<Fallible<DecodedCall>> = batch
+        let calls: Vec<Result<DecodedCall>> = batch
             .iter()
             .map(|call| {
+                if self
+                    .abort_batch
+                    .as_ref()
+                    .map(|b| b.load(Ordering::SeqCst))
+                    .unwrap_or(false)
+                {
+                    return Err(anyhow!("batch aborted"));
```

Comment:
I might actually [bail!](https://docs.rs/anyhow/1.0.31/anyhow/macro.bail.html) instead of `ensure!` here.